### PR TITLE
feat: @talend/utils

### PR DESCRIPTION
**Pull request by sgangnant**
**What is the problem this PR is trying to solve?**
Some utils functions are needed across Talend's products and across @talend/ui packages.

**What is the chosen solution to this problem?**
Create a new `utils` package that contains those utils.
`@talend/utils` is supposed to be lightweight and have as less dependencies as possible.
Content of `@talend/validation` has been moved to `@talend/utils` (and therefore should be deprecated)

**Please check if the PR fulfills these requirements**

* [x] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [x] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/utils/src/date/index.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
+import format from 'date-fns/format';
```
Comment (sgangnant) [inline packages/utils/src/date/index.ts:1]: This was taken from @talend/components and adapted to TypeScript

File: packages/utils/src/validation/index.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import * as REGEXP from './regexp';
```
Comment (sgangnant) [inline packages/utils/src/validation/index.ts:1]: Everything in src/validation comes straight from `@talend/validation`

**Review by jmfrancois — APPROVED** [review]
LGTM

File: packages/utils/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "extends": "talend-scripts extends"
+  },
+  "devDependencies": {
+    "@talend/scripts-core": "^9.11.0",
+    "@talend/scripts-preset-react-lib": "^9.8.2",
+    "@typescript-eslint/parser": "^4.26.1",
+    "cross-env": "^7.0.3"
+  },
+  "dependencies": {
+    "date-fns": "^1.27.2"
```
Comment (gjou-tlnd) [inline packages/utils/package.json:30]: shouldn't we start using date-fns v2 on this new lib?

File: packages/utils/src/date/index.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+	return (timezoneDate.getTime() - utcDate.getTime()) / (1000 * 60);
+}
+
+/**
+ * Ensure a numeric value is expressed on two digits (01, 02, 10 ...)
+ * @param {Number} value
+ * @returns {string}
+ */
+function padTwoDigits(value: number): string {
+	return `${value > 9 ? '' : '0'}${value}`;
```
Comment (gjou-tlnd) [inline packages/utils/src/date/index.ts:47]: https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/String/padStart

Comment (sgangnant) [reply packages/utils/src/date/index.ts:47]: Terrific

**Review by gjou-tlnd — APPROVED** [review]

**Review by romainseb — APPROVED** [review]

**Pull request merged**